### ISSUE TITLE
Support Patch requests

### ISSUE THREAD

**Issue by fenos** [mcustiel/phiremock#14]
Hi, This library is great, thanks! 

However, while I was writing some tests, I was trying to mock a request with the HTTP verb *PATCH* but unfortunately `phiremock` doesn't seem to support this verb. The `A::` class doesn't have the method: `patchRequest` which then I'm unable to mock the call.

Do you have any workaround on this? 
Do you plan to support PATCH HTTP verb as it commonly used in REST to update partially resources?

Thanks a lot!

**Comment by mcustiel** [comment]
Hi @fenos 
Thank you for reporting this, will fix it as soon as I have some free time this week.

**Comment by mcustiel** [comment]
Fixed in version 1.4.0

**Comment by fenos** [comment]
Thanks so much!!

**Comment by fenos** [comment]
@mcustiel Hi, 

I just had a chance to try the `patchRequest` method, unfortunately it throws this exception:

```
[GuzzleHttp\Exception\ServerException] Server error: `POST http://localhost:8086/__phiremock/expectations` resulted in a `500 Internal Server Error` response:
{"result" : "ERROR", "details" : {"request":"method: Field method, was set with invalid value: 'patch'"}}
```

That's how i call it:

```php
$this->phiremock->expectARequestToRemoteServiceWithAResponse(
			Phiremock::on(
                               A::patchRequest()
					->andUrl(Is::equalTo($this->route))
			)->then(
				$response_builder
			)
		);
```

**Comment by mcustiel** [comment]
Hi, @fenos 
Thanks for your comment. 
Have you verified that your installed version of phiremock is the latest (1.4.0)?
For that there is a test in place that does something similar to your code and it is working correctly: 
https://github.com/mcustiel/phiremock/blob/v1.4.0/tests/tests/acceptance/MethodConditionCest.php#L179

**Comment by fenos** [comment]
Mmmh that is interesting, I'm using the codeception extension of Phiremock. I've ran `composer update` and I saw that it pulled down the `v 1.4.0` even the code in the vendor folder matches with the latest changes. 

I can't explain that error :(

**Comment by mcustiel** [comment]
No problem. Will have a look at it this weekend.
Thanks for reporting.

**Comment by mcustiel** [comment]
@fenos I was thinking about this and I guess it can be the cache from the annotations parser. Could you check if in your /tmp directory exists a subdirectory called /phiremock/cache/requests/?
If it does, remove it's contents. If that is the problem I will add a clear cache option to phiremock cli.

**Comment by fenos** [comment]
@mcustiel Hi,

Sorry for the delay, I checked if the folder `/tmp/phiremock/cache/requests` exists but it doesn't.
However, i believe has been deleted once I restarted my laptop or by the OS in some way.
When I run the tests now they are passing even with the `patch` http verb.

I believe was the exact issue you mentioned above, if it doesn't require much effort I suggest to add the deletion of the folder after installing / upgrading the package.
